roofit/roofitcore/inc/RooAbsData.h

```diff
@@ -117,9 +117,19 @@ class RooAbsData : public TNamed, public RooPrintable {
   virtual Double_t weight() const = 0 ; // DERIVED
   virtual Double_t weightSquared() const = 0 ; // DERIVED
   virtual Bool_t valid() const { return kTRUE ; }
+
   enum ErrorType { Poisson, SumW2, None, Auto, Expected } ;
-  virtual Double_t weightError(ErrorType etype=Poisson) const ;
-  virtual void weightError(Double_t& lo, Double_t& hi, ErrorType etype=Poisson) const ; 
+  /// Return the symmetric error on the current weight.
+  /// See also weightError(double&,double&,ErrorType) const for asymmetric errors.
+  /// \param[in] etype Type of error to compute. May throw if not supported.
```

Comment:
Why this needs to be a pure virtual function ?
Not all data set, like the unbinned data set, needs to provide this functionality